next/src/components/_global/nav/nav.tsx

```diff
@@ -0,0 +1,174 @@
+'use client';
+
+import Link from 'next/link';
+import { useEffect, useRef, useState } from 'react';
+import styles from './nav.module.scss';
+import { type Props } from './nav.types';
+import Img from '@/components/ui/Img';
+import Button from '@/components/ui/button/Button';
+
+export default function Nav({ data: { cta }, logo }: Props) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [toggleNav, setToggleNav] = useState(false);
+
+  const navRef = useRef(null);
+  useEffect(() => {
+    const nav = navRef.current as HTMLElement;
+    const navHeight = nav.offsetHeight;
+    let prevScrollPos = window.scrollY;
+    let currentScrollPos = prevScrollPos;
+    let scrollDistance = 0;
+    const handleScroll = () => {
+      prevScrollPos = currentScrollPos;
+      currentScrollPos = window.scrollY;
+      if (currentScrollPos < prevScrollPos && currentScrollPos > navHeight) {
+        nav.classList.add(`${styles.fixed}`, 'fixed');
+        scrollDistance = 0;
+      } else if (nav.classList.contains(styles.fixed)) {
+        scrollDistance += prevScrollPos - currentScrollPos;
+        if (scrollDistance * -1 >= navHeight) {
+          nav.classList.remove(`${styles.fixed}`, 'fixed');
+          scrollDistance = 0;
+        }
+      }
+      if (currentScrollPos === 0) {
+        nav.classList.remove(styles.fixed);
+      }
+    };
+    window.addEventListener('scroll', handleScroll);
+    window.addEventListener('resize', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+      window.removeEventListener('resize', handleScroll);
+    };
+  }, []);
+
+  return (
+    <nav
+      className={styles.navWrapper}
+      ref={navRef}
+    >
+      <div className={`maxWidth ${styles.nav}`}>
+        <Img
```

Comment:
1. Logo should be placed inline as an SVG. 
2. Remember to put the logo in <Link /> and point it to homepage.
3. Remember about accessibility (aria-label)